saleor/graphql/product/bulk_mutations/collection_bulk_delete.py

```diff
@@ -47,6 +47,4 @@ def bulk_action(cls, info, queryset):
         for product in products:
             cls.call_event(manager.product_updated, product, webhooks=webhooks)
 
-        update_products_discounted_prices_for_promotion_task.delay(
-            [product.id for product in products]
-        )
+        get_active_promotion_rules().update(variants_dirty=True)
```

Comment:
This will make ALL existing rules marked as dirty, which means, that we will re-calculate everything.

Reply:
Yep, but we did it already and it was done each time any catalogue have been changed. No, we will mark the rules with flag and we will significantly reduce the number of proceeded promotion rules.
When something have changed in catalogues we have no clue which promotion rules should be marked to recalculate, as rules include filters and we do not know if the given product, variant, category, or collection is included in any of this rule until we filter be the rule predicate